pkg/storage/storage.go

```diff
@@ -575,9 +575,24 @@ func (s *storageModule) ensureCache() error {
 	return nil
 }
 
-// createSubvol creates a subvolume with the given name and limits it to the given size
-// if the requested disk type does not have a storage pool available, an error is
-// returned
+// createSubvolWithQuota creates a subvolume with the given name and limits it to the given size
+// if the requested disk type does not have a storage pool available, an error is returned
+func (s *storageModule) createSubvolWithQuota(size uint64, name string, poolType pkg.DeviceType) (filesystem.Volume, error) {
+	volume, err := s.createSubvol(size, name, poolType)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = volume.Limit(size); err != nil {
+		log.Error().Err(err).Str("volume", volume.Path()).Msg("failed to set volume size limit")
+		return nil, err
+	}
+
+	return volume, nil
+}
+
+// createSubvol creates a subvolume with the given name
+// if the requested disk type does not have a storage pool available, an error is returned
```

Comment:
documentation update is required to state that this size is only for capacity planning (Find a pool that can support the requested size) and that it doesn't limit the size of the subvol to that size